### PR TITLE
Made the following changes:

### PR DESCRIPTION
**Pull request by manikanta-osm**
1. Fixed warnings and notices
2. Change improper variable names
3. Placed hardcoded nonce lifetime variable in config file

### CODE REVIEW

File: UserMetaManagement.php [hunk 1]
```diff
@@ -13,63 +13,55 @@
...
+	wp_register_style('user-meta-font-awesome-css', UMM_PLUGIN_URL . '/css/font-awesome.min.css');
+	wp_register_style('user-meta-datatable-css', UMM_PLUGIN_URL . '/css/datatable-bootstrap.css');
+	wp_register_style('user-meta-style-css', UMM_PLUGIN_URL . '/css/style.css');
+
+	wp_enqueue_style('user-meta-font-awesome-css');
+	wp_enqueue_style('user-meta-datatable-css');
+	wp_enqueue_style('user-meta-style-css');
+	wp_enqueue_style (  'wp-jquery-ui-dialog');
+
+	wp_register_script("user-meta-datatable-js", UMM_PLUGIN_URL . '/js/datatables-min.js', array('jquery'));
```
Comment (AbijeetP) [inline UserMetaManagement.php]: why is this using double quotes?

File: UserMetaManagement.php [hunk 2]
```diff
@@ -89,37 +81,37 @@ public function showMetaForm() {
...
+		    $userId = $userDetails['ID'] == " " ? '--' : $userDetails['ID'];
+		    $userFirstName = ($userMetaInformation['first_name'][0] == '' ? '--' : $userMetaInformation['first_name'][0]);
+		    $userLastName = ($userMetaInformation['last_name'][0] == '' ? '--' : $userMetaInformation['last_name'][0]);
+		    $userEmail = ($userDetailsEmail['user_email'] == '' ? '--' : $userDetailsEmail['user_email']);
+		    $userInformation = array('id' => $userId, 'firstName' => $userFirstName, 'lastName' => $userLastName, 'email' => $userEmail, 'role' => $userRoles);
+		    array_push($user, $userInformation);
+		}
+	    } else {
+		$user = null;
+	    } return ($this->parseTemplate(UMM_TEMPLATE . '/users-same-meta-information.php', $user));
```
Comment (AbijeetP) [inline UserMetaManagement.php]: why is this in the same line?

File: UserMetaManagement.php [hunk 3]
```diff
@@ -178,21 +170,21 @@ public function getUserMetaDetails() {
      */
 
     public function updateUserMetaDetails() {
```
Comment (AbijeetP) [inline UserMetaManagement.php]: add documentation for the methods.

File: UserMetaManagement.php [hunk 4]
```diff
@@ -205,42 +197,44 @@ public function updateUserMetaDetails() {
...
-        echo($this->showMetaForm());
-        echo($this->showAllUsersOfSpecificMetaKey(null, null));
+	echo($this->showMetaForm());
+	echo($this->showAllUsersOfSpecificMetaKey(null, null));
     }
+
     // Fixing the nonce life time to 4 hours.
-     public function nonceLifeTime($time){ 
-        return 60*60*4;
+    public function nonceLifeTime($time) {
```
Comment (AbijeetP) [inline UserMetaManagement.php]: remove 4 hours from constant.

File: UserMetaManagement.php [hunk 5]
```diff
@@ -13,63 +13,55 @@
...
-        wp_enqueue_script('user-meta-script');
-        wp_enqueue_script('user-meta-notify');
-        wp_localize_script('user-meta-script', 'myAjax', array('ajaxurl' => admin_url('admin-ajax.php'), 'ajax_nonce' => wp_create_nonce('user-meta-management')));
+    public function enquerer($hook) {
+	// Checking the page name, if it is not user meta management page, then scripts and styles will not be enqueued
+	if( $hook !== unserialize(UMM_PAGE_NAME)) {
+	    return;
+	}
+	
+	wp_register_style('user-meta-font-awesome-css', UMM_PLUGIN_URL . '/css/font-awesome.min.css');
```
Comment (AbijeetP) [inline UserMetaManagement.php]: please add `umm` infront of all registrations.

File: UserMetaManagement.php [hunk 6]
```diff
@@ -13,63 +13,55 @@
...
-        wp_register_script('user-meta-bootstrap-js', USER_META_PLUGIN_URL . '/js/bootstrap.min.js', array('jquery'));
-        wp_register_script("user-meta-jquery-js", USER_META_PLUGIN_URL . '/js/jquery.js', array('jquery'));
-        wp_register_script("user-meta-datatable-js", USER_META_PLUGIN_URL . '/js/datatables-min.js', array('jquery'));
-        wp_register_script("user-meta-script", USER_META_PLUGIN_URL . '/js/script.js', array('jquery'));
-        wp_register_script("user-meta-notify", USER_META_PLUGIN_URL . '/js/notify.min.js', array('jquery'));
+    public function __construct() {	
+	add_action('admin_enqueue_scripts', array($this, 'enquerer'));
+	add_shortcode('show_meta_form', array($this, 'showMetaForm')); // Shortcode to view the form.
+	add_shortcode('specific_metakey', array($this, 'showAllUsersOfSpecificMetaKey')); // Shortcode to show the users of specific meta key combinations.
+	add_action('wp_ajax_meta_search', array($this, 'metaSearch')); // Action to search the meta key value combinations.
```
Comment (AbijeetP) [inline UserMetaManagement.php]: prefix!

File: UserMetaManagement.php [hunk 7]
```diff
@@ -13,63 +13,55 @@
...
+	add_action('wp_ajax_get_user_meta_details', array($this, 'getUserMetaDetails'));  // Action to show  user meta information to the admin.
+	add_action('wp_ajax_update_user_meta_data', array($this, 'updateUserMetaDetails')); // Action to update the user meta information.
+	add_action('wp_ajax_delete_user_meta', array($this, 'deleteUserMetaDetails')); // Action to delete the user meta information.
```
Comment (AbijeetP) [inline UserMetaManagement.php]: comments on top of the statement.

File: css/style.css [hunk 8]
```diff
@@ -161,18 +144,248 @@
...
+.table-striped>tbody>tr:nth-of-type(odd) {
+    background-color: #f9f9f9;
+}
+div#wpwrap {
+    background: #fff;
+}
+.table-striped > tbody > tr {
+    border-bottom: 1px #ddd;
+}
+thead {
```
Comment (AbijeetP) [inline css/style.css]: add prefix for classes as well.

File: css/style.css [hunk 9]
```diff
@@ -161,18 +144,248 @@
...
+table td {
+    padding: 8px;
+}
+table th, table td {
+    border: none;
+}
+.meta-form .form-inline .form-group {
+    display: inline-block;
+}
+.btn {
```
Comment (AbijeetP) [inline css/style.css]: prefix for buttons also!

File: js/script.js [hunk 10]
```diff
@@ -1,11 +1,56 @@
...
     var userId;
+       
+    $(".dialog-info, .dialog-warning, .dialog-danger, .dialog-success").dialog({
+	autoOpen: false,
+	modal:true,
+	open: function () {
+	    designAlertDialogBoxes();
+	}
+    });   
+    $(".close-dialog").click(function (){
```
Comment (AbijeetP) [inline js/script.js]: please add a new line before this.

File: js/script.js [hunk 11]
```diff
@@ -1,11 +1,56 @@
...
+	metaFormDialog.dialog('open');
+    });
+    
     $('#usersMeta').dataTable({
         "dom": '<"top">rt<"bottom" pl>',
         "order": [[0, "asc"]],
         'scrollX': true,
-        "autoWidth": false
+        "autoWidth": false,
+	"columnDefs": [
```
Comment (AbijeetP) [inline js/script.js]: code is not formatted.

File: js/script.js [hunk 12]
```diff
@@ -42,10 +87,10 @@ var userMetaManager = (function () {
             }
         }
         var data = {
-            'action': 'update_user_meta_data',
-            'security': myAjax.ajax_nonce,
-            'userId': userId,
-            'userMetaData': updatedMetaData
+            action: 'update_user_meta_data',
```
Comment (AbijeetP) [inline js/script.js]: write some comments on top of the loops to explain what you are doing.

File: js/script.js [hunk 13]
```diff
@@ -60,28 +105,45 @@ var userMetaManager = (function () {
         deleteUserMetaInformation();
     });
 
+    /**
+     * Method to add styles to the alert message dialog boxes.
+     */
+    function designAlertDialogBoxes() {
```
Comment (AbijeetP) [inline js/script.js]: please get the reference to the currently opened dialog and modify only that dialog.

File: js/script.js [hunk 14]
```diff
@@ -123,7 +184,7 @@ var userMetaManager = (function () {
             "paging": false,
             "scrollY": "400px",
             "scrollCollapse": false,
-            sortable: false,
+             sortable: false,
```
Comment (AbijeetP) [inline js/script.js]: please remove " from the code.

File: template/users-same-meta-information.php [hunk 15]
```diff
@@ -5,11 +5,11 @@
     <table id="usersMeta" class="table table-striped">
         <thead>
             <tr>
-                <th>ID</th>
-                <th>First name</th>
-                <th>Last name</th>
-                <th>Email</th>
-                <th>Role</th>
+		<th>Email<i class="fa fa-fw fa-sort"></i></th>
```
Comment (AbijeetP) [inline template/users-same-meta-information.php:8]: format code.

**Review by AbijeetP — CHANGES_REQUESTED** [review]
Please make the discussed changes.

File: template/users-same-meta-information.php (hunk 15 above)
Comment (manikanta-osm) [reply template/users-same-meta-information.php:8]: Okay Abijeet

File: UserMetaManagement.php [hunk 16]
```diff
@@ -214,23 +216,23 @@ public function deleteUserMetaDetails() {
     }
 
     // Function to create a user page in the user secion in the user seciton.
-    public function addUserPage() {
-	add_users_page('User Meta Management', 'User Meta Management', 'manage_options', 'user-meta-management', array($this, 'userMetaManagement'));
+    public function UMMAddUserPage() {
```
Comment (AbijeetP) [inline UserMetaManagement.php]: please dont prefix function names. They are inside a class ... so it is not an issue that they will ever clash.

**Review by AbijeetP — CHANGES_REQUESTED** [review]

**Review by AbijeetP — APPROVED** [review]

**Pull request merged**